pkg/cmd/extension/command.go

```diff
@@ -132,6 +127,9 @@ func NewCmdExtension(f *cmdutil.Factory) *cobra.Command {
 					if len(args) > 0 && flagAll {
 						return cmdutil.FlagErrorf("cannot use `--all` with extension name")
 					}
+					if flagForce && flagDryRun {
+						return cmdutil.FlagErrorf("cannot use `--force` and `--dry-run`")
```

Comment:
Why not allowing force with dry-run? The dry run could print information that the upgrade will be forced, i.e. that the local git checkout will be replaced with the contents of the remote.